### ISSUE TITLE
Clarity on the design process for VB and C#

### ISSUE THREAD

**Issue by AdamSpeight2008** [dotnet/roslyn#24470]
Both the VB and C# language repositories have a read me document that outlines the process for language features. I've extracted out two sections that I think did further clarity.

>3. A prototype owner (who may or may not be proposal owner) should implement a prototype in their own fork of the Roslyn repo and share it with the design team and community for feedback. A prototype must meet the following bar:
>
> * Parsing (if applicable) should be resilient to experimentation--typing should not cause crashes.
> * Include minimal tests demonstrating the feature at work end-to-end.
> * Include minimal IDE support (keyword coloring, formatting, completion).
>
>4. Once a prototype has proven out the proposal and the proposal has been approved-in-principle by the design team, a feature owner (who may or may not be proposal or prototype owner(s)) implemented in a feature branch of the Roslyn repo. The bar for implementation quality can be found here.

Especially the following part of `3` 
 > share it with the design team and community for feedback

How can they share it with the design team and community for feedback?
Submitting a PR to the Roslyn repo, seems to go against section 4. 
Feedback could be a useful especially if the need eyes and helping hands of their communities to overcome tricky parts. They may just be out of reach for the current level of knowledge.

**Comment by gafter** [comment]
You share a prototype for feedback by pointing people to a fork of the Roslyn repository. Microsoft employees tend to use a git branch on the dotnet/roslyn repo, but you can use a branch in your own repo.

**Comment by gafter** [comment]
The kind of feedback step 3 of this process is intended to support is feedback about the proposed language feature as designed, not the code that implements its prototype. Submitting a PR would not solicit feedback about the language feature, but rather solicits feedback about its implementation, which is not what should be happening in step 3.

**Comment by gafter** [comment]
@jaredpar Do you want to add anything about how an external contributor can prepare a usable prototype of a proposed language feature? Is there a documented way to do that?

**Comment by AdamSpeight2008** [comment]
@gafter @jaredpar Sorry if I've been a pain on these points, it just frustrating *(probably on both sides)* that I feel really excited to be close to getting the feature working *(at least to my low standards)*. And the key or help to reach it, is not forthcoming. *I understand that it isn't an automatic right, but the occasional scraps from table of the languages gods would be treasured and respected.*

**Comment by jaredpar** [comment]
@AdamSpeight2008 

> How can they share it with the design team and community for feedback?

I think the best way to do this is to start a discussion on the appropriate language repo, provide a speclet of the feature and link to the implementation branch.

> Submitting a PR to the Roslyn repo, seems to go against section 4.

Correct. That's creating a PR that has almost 0 chance of being merged. 

This is not a limitation we apply to the community, it's one we apply to ourselves as well. Language features are **hard**. They don't just encompass compiler changes but have to take into account every aspect of the language experience: Intellisense, coloring, API shape, ENC, debugging, etc ... It usually takes many iterations and contributions from developers who are knowledgable in those specific areas to complete a feature. 

This is why we typically don't do PRs for language features into our release branches. It's simply not realistic to complete a language feature in a single PR. Merging a partially completed feature into a release branch is not acceptable because our mentality is to have these branches in an "always ready to ship" state. Partial features go against this. 

> And the key or help to reach it, is not forthcoming

Adam we've provided the feedback and documentation on how to approach these problems for some time now. Several of the docs you've referenced were created specifically in reaction to PRs you sent in order to give clarity to you and the community. 

Yet no matter how many times we provide certain pieces of feedback or guidance you simply do not follow it. I've lost track of the number of times we've had to ask you to:

1. Stop including large code formatting changes in PR. 
1. Stop sending PRs for surprise language features and asking for review.
1. Stop opening a PR, quickly closing a PR and then opening a new PR for the same change. 

I'm not sure what the disconnect is here but we feel we've been **extremely** clear on this points. I understand you don't agree with these, in particular the code formatting, but those are the rules by which we run this repository.
